### PR TITLE
Added validate_granule_data_format_against_collection_check and granule_data_format_presence_check

### PR DESCRIPTION
**Pull request by jenny-m-wood**
The validate_granule_data_format_against_collection_check is intended to check the data format provided in the granule (both echo10 and umm-g) matches the data format provided in the collection. The granule_data_format_presence_check is intended to check that a data format is provided in the granule metadata (both echo10 and umm-g).

### CODE REVIEW

File: pyQuARC/schemas/checks.json [hunk 1]
```diff
@@ -234,9 +234,19 @@
...
         "check_function": "get_data_url_check",
         "available": true
+    },
+    "granule_data_format_presence_check": {
+        "data_type": "custom",
+        "check_function": "presence_check",
+        "available": true
     }
-}
+}
```
Comment (svbagwell) [inline pyQuARC/schemas/checks.json:252]:
```suggestion
}

```
I think it's good practice to include a new line at the end of the file

File: pyQuARC/schemas/check_messages.json [hunk 2]
```diff
@@ -926,5 +926,21 @@
...
+    "granule_data_format_presence_check": {
+        "failure": "The granule data format is missing.",
+        "help": {
+            "message": "",
+            "url": ""
+        },
+        "remediation": "Recommend providing the granule data format."
     }
-}
+}
```
Comment (svbagwell) [inline pyQuARC/schemas/check_messages.json:946]:
```suggestion
}

```
I think it's good practice to include a new line at the end of a file

File: pyQuARC/schemas/rule_mapping.json [hunk 3]
```diff
@@ -4591,5 +4599,53 @@
...
+                        "Granule/DataFormat"
+                    ]
+                }
+            ]
+        },
+        "severity": "warning",
+        "check_id": "granule_data_format_presence_check"
     }
-}
+}
```
Comment (svbagwell) [inline pyQuARC/schemas/rule_mapping.json:4651]:
```suggestion
}

```
I think it's good practice to add a new line to the end of a file

**Review by svbagwell — COMMENTED** [review]
LGTM

**Review by svbagwell — APPROVED** [review]

**Review by esr0004 — COMMENTED** [review]
For the "Validate Granule Data Format Against Collection Check" (validate_granule_data_format_against_collection_check), recommend changing the fields under echo-g in the rule_mapping.json to the following:

“Granule/DataFormat”
“Granule/Collection/ShortName”
“Granule/Collection/VersionId”
“Granule/Collection/DataSetId”

**Review by esr0004 — APPROVED** [review]

**Pull request merged**